CRM/Core/Payment/AuthorizeNetIPN.php

```diff
@@ -41,13 +41,6 @@ public function __construct($inputData) {
    */
   public function main() {
     try {
-      //we only get invoice num as a key player from payment gateway response.
-      //for ARB we get x_subscription_id and x_subscription_paynum
-      $x_subscription_id = $this->retrieve('x_subscription_id', 'String');
```

Comment:
This retrieve function throws an exception if not retrieved - so the check is implict & we are already doing it by retrieving it a little later